File: java/org/mcw/monstercapitalworks/model/PlayerData.java
package org.mcw.monstercapitalworks.model;

import org.bukkit.entity.EntityType;

import java.util.HashMap;
import java.util.HashSet;
import java.util.Map;
import java.util.Set;
import java.util.UUID;

/**
 * 玩家数据模型
 */
public class PlayerData {
    private final UUID uuid;
    private int normalPurchasedLimit;
    private int premiumPurchasedLimit;
    private final Map<SpawnerType, Set<EntityType>> unlockedEntities;

    public PlayerData(UUID uuid) {
        this.uuid = uuid;
        this.normalPurchasedLimit = 0;
        this.premiumPurchasedLimit = 0;
        this.unlockedEntities = new HashMap<>();
        this.unlockedEntities.put(SpawnerType.NORMAL, new HashSet<>());
        this.unlockedEntities.put(SpawnerType.PREMIUM, new HashSet<>());
    }

    public UUID getUuid() {
        return uuid;
    }

    public int getNormalPurchasedLimit() {
        return normalPurchasedLimit;
    }

    public void setNormalPurchasedLimit(int normalPurchasedLimit) {
        this.normalPurchasedLimit = normalPurchasedLimit;
    }

    public int getPremiumPurchasedLimit() {
        return premiumPurchasedLimit;
    }

    public void setPremiumPurchasedLimit(int premiumPurchasedLimit) {
        this.premiumPurchasedLimit = premiumPurchasedLimit;
    }
    
    /**
     * 获取指定类型的已购买限制
     */
    public int getPurchasedLimit(SpawnerType type) {
        return type == SpawnerType.NORMAL ? normalPurchasedLimit : premiumPurchasedLimit;
    }
    
    /**
     * 设置指定类型的已购买限制
     */
    public void setPurchasedLimit(SpawnerType type, int limit) {
        if (type == SpawnerType.NORMAL) {
            this.normalPurchasedLimit = limit;
        } else {
            this.premiumPurchasedLimit = limit;
        }
    }
    
    /**
     * 增加已购买限制
     */
    public void addPurchasedLimit(SpawnerType type, int amount) {
        if (type == SpawnerType.NORMAL) {
            this.normalPurchasedLimit += amount;
        } else {
            this.premiumPurchasedLimit += amount;
        }
    }

    public Set<EntityType> getUnlockedEntities(SpawnerType type) {
        return unlockedEntities.get(type);
    }

    public boolean hasUnlockedEntity(SpawnerType type, EntityType entityType) {
        return unlockedEntities.get(type).contains(entityType);
    }

    public void unlockEntity(SpawnerType type, EntityType entityType) {
        unlockedEntities.get(type).add(entityType);
    }

    public void lockEntity(SpawnerType type, EntityType entityType) {
        unlockedEntities.get(type).remove(entityType);
    }
}

File: MonsterCapitalWorks/src/main/kotlin/org/mcw/monstercapitalworks/model/PlayerData.kt
package org.mcw.monstercapitalworks.model

import org.bukkit.entity.EntityType
import java.util.*

/**
 * 玩家数据模型
 */
data class PlayerData(
    val uuid: UUID,
    var normalPurchasedLimit: Int = 0,
    var premiumPurchasedLimit: Int = 0,
    private val unlockedEntities: MutableMap<SpawnerType, MutableSet<EntityType>> = mutableMapOf(
        SpawnerType.NORMAL to mutableSetOf(),
        SpawnerType.PREMIUM to mutableSetOf()
    )
) {
    /**
     * 获取指定类型的已购买限制
     */
    fun getPurchasedLimit(type: SpawnerType): Int =
        if (type == SpawnerType.NORMAL) normalPurchasedLimit else premiumPurchasedLimit

    /**
     * 设置指定类型的已购买限制
     */
    fun setPurchasedLimit(type: SpawnerType, limit: Int) {
        when (type) {
            SpawnerType.NORMAL -> normalPurchasedLimit = limit
            SpawnerType.PREMIUM -> premiumPurchasedLimit = limit
        }
    }

    /**
     * 增加已购买限制
     */
    fun addPurchasedLimit(type: SpawnerType, amount: Int) {
        when (type) {
            SpawnerType.NORMAL -> normalPurchasedLimit += amount
            SpawnerType.PREMIUM -> premiumPurchasedLimit += amount
        }
    }

    fun getUnlockedEntities(type: SpawnerType): Set<EntityType> =
        unlockedEntities[type] ?: emptySet()

    fun hasUnlockedEntity(type: SpawnerType, entityType: EntityType): Boolean =
        unlockedEntities[type]?.contains(entityType) ?: false

    fun unlockEntity(type: SpawnerType, entityType: EntityType) {
        unlockedEntities[type]?.add(entityType)
    }

    fun lockEntity(type: SpawnerType, entityType: EntityType) {
        unlockedEntities[type]?.remove(entityType)
    }
}
